examples/i-inverse-kinematics.py

```diff
@@ -1,27 +1,31 @@
-import numpy
+import numpy as np
 import pinocchio
-from sys import argv
 
 model = pinocchio.buildSampleModelManipulator()
 data  = model.createData()
 
 JOINT_ID = 6
-DT       = 1e-1
+xdes     = np.matrix([ 0.5,-0.5,0.5]).T
+
 q        = model.neutralConfiguration
-xdes     = numpy.matrix([ 0.5,-0.5,0.5]).T
+eps      = 1e-4
+IT_MAX   = 1000
+DT       = 1e-1
 
-for i in range(100):
-    pinocchio.computeJointJacobians(model,data,q)
-    J   = pinocchio.jointJacobian(model,data,q,JOINT_ID,pinocchio.ReferenceFrame.LOCAL,True)[:3,:]
+for i in range(IT_MAX):
+    pinocchio.forwardKinematics(model,data,q)
     x   = data.oMi[JOINT_ID].translation
     R   = data.oMi[JOINT_ID].rotation
     err = R.T*(x-xdes)
-    v   = - numpy.linalg.pinv(J)*err
+    if np.linalg.norm(err) < eps:
+        print "Convergence achieved!"
+        break
+    J   = pinocchio.jointJacobian(model,data,q,JOINT_ID,pinocchio.ReferenceFrame.LOCAL,True)[:3,:]
+    v   = - np.linalg.pinv(J)*err
     q   = pinocchio.integrate(model,q,v*DT)
     if not i % 10:        print 'error = ', (x-xdes).T
+else:
+    print "\nWarning: the iterative algorithm has not reached convergence to the desired precision"
 
-# Computing error for final q
-pinocchio.forwardKinematics(model,data,q)
-x   = data.oMi[JOINT_ID].translation
-err = x-xdes
-print '\nfinal error: ', err.T
+print '\nresult: ', q.flatten().tolist()
+print '\nfinal error: ', (x-xdes).T
```

Comment:
please use the print function instead of the statement, is this statement is not valid in Python 3.

To do so: add `from __future__ import print_function` before anything else in the file, and then use it with for example with `print('\nfinal error: %s' % (x-xdes).T)`

Reply:
I also prefer the function syntax. In fact, I had considered using it, but then I did not do it because I did not want to add things which are not strictly necessary to the example. But I can change it, in fact it is much nicer. We just have to agree we will consistently use it throughout all of our examples.

One thing I have never tried/understood: what happens if you execute a file containing `from __future__ import print_function` in Python 3? I guess this statement is simply ignored?